util/dvsim/NcLauncher.py

```diff
@@ -150,14 +157,51 @@ def poll(self):
         """
 
         assert self.process is not None
-        elapsed_time = datetime.datetime.now() - self.start_time
-        job_runtime_secs = elapsed_time.total_seconds()
         if self.process.poll() is None:
-            timeout_mins = self.deploy.get_timeout_mins()
-            if timeout_mins is not None and not self.deploy.gui:
-                if job_runtime_secs > (timeout_mins * 60):
+            run_timeout_mins = self.deploy.get_timeout_mins()
+            wait_timeout_mins = 180  # max wait time in job queue / license
+            file_size_thresh_bytes = 5000  # log file size threshold
+            if run_timeout_mins is not None and not self.deploy.gui:
+                job_waittime = 0
+                job_runtime = 0
+                # query the log file size
+                f_size = os.path.getsize(self.deploy.get_log_path())
+                if f_size < file_size_thresh_bytes:
+                    # if the job log size is less than the threshold, increment
+                    # the job wait time counter. hold the run time counter at 0
+                    self.start_run_time = 0
+                    elapsed_wait_time = (datetime.datetime.now() -
+                                         self.start_time)
+                    job_waittime = elapsed_wait_time.total_seconds() / 60
+                else:
+                    # if the job log size is more than the threshold, start the
+                    # the job run time counter if has not already started.
+                    # Hold the wait time counter at the last updated value
+                    if (self.start_run_time == 0):
+                        self.start_run_time = datetime.datetime.now()
+                    elapsed_wait_time = (self.start_run_time -
+                                         self.start_time)
+                    job_waittime = elapsed_wait_time.total_seconds() / 60
+                # if the job run time counter has already started, increment
+                # the job run time counter
+                if (self.start_run_time != 0):
+                    elapsed_run_time = (datetime.datetime.now() -
+                                        self.start_run_time)
+                    job_runtime = elapsed_run_time.total_seconds() / 60
+                # if the job run has not started within the max time specified
+                # in this file or if the job has been running for longer than
+                # the time specified in the block json or via dvsim switches
+                # terminate it and send the appropriate error message
+                wait_timeout = (job_waittime > wait_timeout_mins)
+                run_timeout = (job_runtime > run_timeout_mins)
+                if run_timeout or wait_timeout:
```

Comment:
I believe this is correct, but it's pretty hard to read. Couldn't this be slightly restructured, to be a shape like this?
```python
  timeout_mins = None
  if self.start_run_time != 0:
    timeout_mins = run_timeout_mins
    phase_start_time = self.start_run_time
  elif self.start_wait_time != 0:
    timeout_mins = wait_timeout_mins
    phase_start_time = self.start_wait_time

  if timeout_mins is not None:
    elapsed = datetime.datetime.now() - phase_start_time
    elapsed_mins = elapsed.total_seconds() / 60
    if elapsed_mins > timeout_mins:
      timeout_message = '....'
      ...
```

Reply:
I didn't follow this approach as I had to change the logic to accommodate your other comments. Can you please check my new patch and let me know if I still need to reorder ?